### PR TITLE
[serve] Add basic REST API to dashboard

### PR DESCRIPTION
**Pull request by edoakes**
First run `ray start --head`

Then deploy something:
`python test.py`
```python
import ray
ray.init(address="auto")

from ray import serve

serve.start(detached=True)

@serve.deployment
def f(): pass

f.deploy()
```

Then go to `http://localhost:8265/api/serve/deployments/`, output is:
`{"f": "Deployment(name=f,version=None,route_prefix=/f)"}`

### CODE REVIEW

**Comment by edoakes** [comment]
@shrekris-anyscale this is more fully-fledged now. Probably still a lot of issues :)

**Comment by edoakes** [comment]
@shrekris-anyscale can you stamp so i can merge please?

**Review by shrekris-anyscale — APPROVED** [review]

File: dashboard/optional_utils.py [hunk 1]
```diff
@@ -242,3 +249,51 @@ def _update_cache(task):
...
+                            _redis_password=redis_pw,
+                        )
+                    except Exception as e:
+                        ray.shutdown()
+                        raise e from None
+
+                if connect_to_serve:
+                    # TODO(edoakes): this should probably run in the `serve`
+                    # namespace.
+                    serve.start(detached=True)
```
Comment (shrekris-anyscale) [inline dashboard/optional_utils.py:287]: Are the deployments deployed correctly without starting Serve in the dashboard's namespace?

Comment (edoakes) [reply dashboard/optional_utils.py:287]: We need to fix this more generally though so things don't start in this wonky namespace

**Pull request merged**